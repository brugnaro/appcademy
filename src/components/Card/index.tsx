import React from 'react';
import {Text, View} from 'react-native';

import {Computer} from '../../assets/icons';

import styles from './styles';

const Card = (props: {subtitle: String, title: String}) => (
  <View style={styles.container}>
    <Computer />
    <View style={styles.content}>
      <Text style={styles.title}>{props.title}</Text>
      <Text style={styles.subtitle}>{props.subtitle}</Text>
    </View>
  </View>
);

export default Card;
