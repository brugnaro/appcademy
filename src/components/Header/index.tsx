import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';

import {Back} from '../../assets/icons';

import styles from './styles';

const Header = (props: {goBack: any; title: String}) => (
  <View style={styles.container}>
    <TouchableOpacity onPress={props.goBack}>
      <Back />
    </TouchableOpacity>
    <Text style={styles.text}>{props.title}</Text>
    <View />
  </View>
);

export default Header;
