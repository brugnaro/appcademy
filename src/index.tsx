import React from 'react';

import {Store} from './store';

import {MAIN_INITIAL_STATE} from './store/main/reducers';

import reducers from './store/reducers';

import Routes from './routes';

const App = () => {
  const initialState = {
    main: MAIN_INITIAL_STATE,
  };
  return (
    <Store initialState={initialState} reducer={reducers}>
      <Routes />
    </Store>
  );
};
export default App;
