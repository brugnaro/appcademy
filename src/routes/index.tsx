import React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {NavigationContainer} from '@react-navigation/native';

const Main = createNativeStackNavigator();

import {Home, List} from '../screens';

const Routes: React.FC = () => (
  <NavigationContainer>
    <Main.Navigator
      screenOptions={{
        contentStyle: {backgroundColor: '#f6f6f6'},
        headerShown: false,
      }}>
      <Main.Screen name="Home" component={Home} />
      <Main.Screen name="Lista" component={List} />
    </Main.Navigator>
  </NavigationContainer>
);

export default Routes;
