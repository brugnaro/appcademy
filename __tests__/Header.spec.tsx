import React from 'react';
import {create} from 'react-test-renderer';

import {Header} from '../src/components';

describe('Testing Header component', () => {
  it('renders correctly', () => {
    const tree = create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Test title props', () => {
    const component = create(<Header title="Home" />).root;
    expect(component.props.title).toBe('Home');
  });
});
