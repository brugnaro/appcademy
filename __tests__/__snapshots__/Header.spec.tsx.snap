// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#215897",
      "borderBottomColor": "#F8D117",
      "borderBottomWidth": 4,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="18"
      bbWidth="18"
      focusable={false}
      height="18"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 18,
            "width": 18,
          },
        ]
      }
      vbHeight={490.787}
      vbWidth={490.787}
      width="18"
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M362.671,490.787c-2.831,0.005-5.548-1.115-7.552-3.115L120.452,253.006 c-4.164-4.165-4.164-10.917,0-15.083L355.119,3.256c4.093-4.237,10.845-4.354,15.083-0.262c4.237,4.093,4.354,10.845,0.262,15.083 c-0.086,0.089-0.173,0.176-0.262,0.262L143.087,245.454l227.136,227.115c4.171,4.16,4.179,10.914,0.019,15.085 C368.236,489.664,365.511,490.792,362.671,490.787z"
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M362.671,490.787c-2.831,0.005-5.548-1.115-7.552-3.115L120.452,253.006c-4.164-4.165-4.164-10.917,0-15.083L355.119,3.256 c4.093-4.237,10.845-4.354,15.083-0.262c4.237,4.093,4.354,10.845,0.262,15.083c-0.086,0.089-0.173,0.176-0.262,0.262 L143.087,245.454l227.136,227.115c4.171,4.16,4.179,10.914,0.019,15.085C368.236,489.664,365.511,490.792,362.671,490.787z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 18,
        "paddingRight": 20,
      }
    }
  />
  <View />
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#215897",
      "borderBottomColor": "#F8D117",
      "borderBottomWidth": 4,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="18"
      bbWidth="18"
      focusable={false}
      height="18"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 18,
            "width": 18,
          },
        ]
      }
      vbHeight={490.787}
      vbWidth={490.787}
      width="18"
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M362.671,490.787c-2.831,0.005-5.548-1.115-7.552-3.115L120.452,253.006 c-4.164-4.165-4.164-10.917,0-15.083L355.119,3.256c4.093-4.237,10.845-4.354,15.083-0.262c4.237,4.093,4.354,10.845,0.262,15.083 c-0.086,0.089-0.173,0.176-0.262,0.262L143.087,245.454l227.136,227.115c4.171,4.16,4.179,10.914,0.019,15.085 C368.236,489.664,365.511,490.792,362.671,490.787z"
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M362.671,490.787c-2.831,0.005-5.548-1.115-7.552-3.115L120.452,253.006c-4.164-4.165-4.164-10.917,0-15.083L355.119,3.256 c4.093-4.237,10.845-4.354,15.083-0.262c4.237,4.093,4.354,10.845,0.262,15.083c-0.086,0.089-0.173,0.176-0.262,0.262 L143.087,245.454l227.136,227.115c4.171,4.16,4.179,10.914,0.019,15.085C368.236,489.664,365.511,490.792,362.671,490.787z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 18,
        "paddingRight": 20,
      }
    }
  />
  <View />
</View>
`;
